cmd/apiserver/app/server/run_server.go

```diff
@@ -35,7 +36,7 @@ func RunServer(opts *ServiceCatalogServerOptions, stopCh <-chan struct{}) error
 	if stopCh == nil {
 		/* the caller of RunServer should generate the stop channel
 		if there is a need to stop the API server */
-		stopCh = make(chan struct{})
+		stopCh = server.SetupSignalHandler()
```

Comment:
should check the callers of this and make sure they're supplying a channel. 

My guess is that there is the cli/cmd , and tests.

We should probably crash if this isn't set, but I haven't looked close enough to know.

Reply:
i will have a look about this. thank you for your review @MHBauer

Reply:
Have you looked into this? what happens when it isn't set?

Reply:
sorry, i was busy these days. i Will have a look it today or tomorrow.

Reply:
@jberkhahn the caller give a channel actually, which created by server.SetupSignalHandler(). 
but if stopCh == nil,  just make one can not be closed at all. so i give the code above.

Reply:
from that description, I think this should panic and crash with an informative message.

Reply:
@MHBauer i modify the code to "panic", also give a message when the stop channel was not set. Please help review about this when you are idle.